src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -2285,5 +2285,58 @@ private static uint Int64DivMod1E9(ref ulong value)
             value /= 1000000000;
             return rem;
         }
+
+        private static unsafe void DoubleToNumber(double value, int precision, ref NumberBuffer number)
+        {
+            number.precision = precision;
+
+            if (double.GetExponent(value) == 0x7FF)
```

Comment:
Should this rather be `double.IsFinite` ?

Reply:
Fixed.